frontend/src/pages/sport items/teamPage.js

```diff
@@ -0,0 +1,161 @@
+import SearchBlock from "../../components/containers/searchBlock.jsx";
+import NoItems from "../../components/NoItems.js";
+import Filters from "../../components/containers/filtersBlock.jsx";
+import { RiArrowLeftWideLine } from "react-icons/ri";
+
+function TeamPage() {
+    const { leagueName } = useParams();
+
+    const cardSizes = {
+        large: { rows: 4, columns: 4, cardSize: { width: 280, height: 350 }, postsPerPage: 16 },
+        medium: { rows: 5, columns: 5, cardSize: { width: 220, height: 280 }, postsPerPage: 25 },
+        small: { rows: 10, columns: 2, cardSize: { width: 600, height: 100 }, postsPerPage: 20 }
+    };
+
+    const navigate = useNavigate();
+    const location = useLocation();
+    const stateData = location.state || {};
+    const leagueId = stateData.leagueId;
+    const sportId = stateData.sportId;
+
+    const [currentTeams, setCurrentTeams] = useState([]);
+    const [pageCount, setPageCount] = useState(0);
+    const [paginationKey, setPaginationKey] = useState(0);
+    const [currentPage, setCurrentPage] = useState(0);
+    const [teamsPerPage, setTeamsPerPage] = useState(cardSizes.large.postsPerPage);
+    const [gridSize, setGridSize] = useState(cardSizes.large);
+
+    const [loading, setLoading] = useState(false);
+    const [inputValue, setInputValue] = useState('');
+    const [prevInputValue, setPrevInputValue] = useState('');
+    const [searchClicked, setSearchClicked] = useState(false);
+    const [passedPosts, setPassedPosts] = useState(0);
+
+    useEffect(() => {
+        let page = Math.floor(passedPosts / teamsPerPage);
+        setCurrentPage(page);
+        getTeams(page);
+    }, [teamsPerPage]); 
+
+    const handleGridSizeChange = (size) => {
+        console.log('size: ', size);
+        if (cardSizes[size]) {
+            setPassedPosts(gridSize.rows * gridSize.columns * currentPage);
+            setTeamsPerPage(cardSizes[size].postsPerPage);
+            setGridSize(cardSizes[size]);
+        } else {
+            setGridSize(cardSizes.large);
+        }
+    };
+
+
+    useEffect(() => {
+        getTeams(0);
+    }, []);
+
+    const handlePageClick = (event) => {
+        const selectedPage = event.selected;
+        setCurrentPage(selectedPage);
+        getTeams(selectedPage);
+        window.scrollTo({ top: 0, behavior: "smooth" });
+    };
+
+    function handleSearchClick() {
+        setSearchClicked((prev) => !prev);
+    }
+
+    const handleKeyDown = (event) => {
+        if (event.key === 'Enter') {
+            handleSearchClick();
+        }
+    };
+    const getTeams = async (page) => {
+
+        setPrevInputValue(inputValue);
+
+        try {
+            setLoading(true);
+
+            const response = await axios.post(
+                `${apiEndpoints.url}${apiEndpoints.teams.getAll}`,
+                {
+                    teams__sport_id: sportId,
+                    leagues__league_id: leagueId,
+                    //  letter: inputValue,
```

Comment:
Comment again

Either extend the cooment and specify what that thing does and when it might be usefult to have it uncommented or just remove